Sources/KlaviyoUI/InAppForms/Models/IAFNativeBridgeEvent.swift

```
@@ -47,7 +47,14 @@ enum IAFNativeBridgeEvent: Decodable, Equatable {
             let data = try JSONEncoder().encode(decodedData)
             self = .trackProfileEvent(data)
         case .openDeepLink:
-            self = .openDeepLink
+            let url = try container.decode(DeepLinkEventPayload.self, forKey: .data)
+            self = .openDeepLink(url.ios)
         }
     }
 }
+
+extension IAFNativeBridgeEvent {
+    struct DeepLinkEventPayload: Codable {
+        let ios: URL
```

we are receiving both `ios` and `android` from the payload but imagine we only care about iOS so it'll just contain this

---

What'll happen here if the string is not a valid URL?

---

@evan-masseau was this going to change based on the SDK name that gets injected into JS?

---

I hadn't considered that, but yes i suppose it could. Back to the event spec doc I guess

---

if it's not a valid URL it would throw an error when we try to decode it to this type (line 50) -- I can add safer handling to it so it just silently exits. I also assumed we were doing validation on the message bus side, but if we want to just treat it as a string and safely cast it to a URL on this end we could

---

Yeah lets put in safer handling here, as far as I know, right now there's no point earlier in the system that's performing URI validation on this, and regardless its best to validate at each level.

---

@belleklaviyo what were you thinking in terms of safer handling? The way it's currently written, any time we try to decode data into an `IAFNativeBridgeEvent` type, we're forced to handle a potential decoding error (like [here](https://github.com/klaviyo/klaviyo-swift-sdk/blob/d1f0453eeea98cfdbdabd327f79e095a2c648e87/Sources/KlaviyoUI/InAppForms/IAFWebViewModel.swift#L47), for example). So I think we already do safely handle a decoding error

---

but we don't validate URL anywhere no? It just probably won't call the delegate method. There is a method called [canOpenURL](https://developer.apple.com/documentation/uikit/uiapplication/canopenurl(_:)) which we could use.

---

> if it's not a valid URL it would throw an error when we try to decode it to this type (line 50)

Forgive my swift ignorance, will that inline `try` on line 50 throw an error or fail silently? I just don't want to introduce a potential crash of the host app. But if that is handled, then ignore what I said!

---

Good catch @evan-masseau. @belleklaviyo might want to wrap that in a do-catch block.

---

nvm looks like she's already aware of that - https://github.com/klaviyo/klaviyo-swift-sdk/pull/269/files#r1943703046

---

@belleklaviyo sorry, I'm not sure I agree with the change you made. I think you can keep the associated value as a `URL` type. We already safely handle the case where it's not a valid URL (see [my comment above](https://github.com/klaviyo/klaviyo-swift-sdk/pull/269#discussion_r1943712814)). 

>Forgive my swift ignorance, will that inline try on line 50 throw an error or fail silently?

@evan-masseau, it throws an error and we're forced to handle it

---

ahh okay yeah sure, I'll still also add in the `canOpenURL` still though as an extra check cause it seems neat (I didn't know that existed lol)